datasync/chaindatafetcher/kafka/consumer.go

```diff
@@ -101,20 +158,130 @@ func (c *Consumer) Cleanup(s sarama.ConsumerGroupSession) error {
 	return nil
 }
 
+// insertSegment inserts the given segment to the given buffer.
+// Assumption:
+//  1. it is guaranteed that the order of segments is correct.
+//  2. the inserted messages may be duplicated.
+//
+// We can consider the following cases.
+// case1. new segment with index 0 is inserted into newly created segment slice.
+// case2. new consecutive segment is inserted into the right position.
+// case3. duplicated segment is ignored.
+// case4. new sparse segment shouldn't be given, so return an error.
+func insertSegment(segment *Segment, buffer [][]*Segment) ([][]*Segment, error) {
+	for idx, segments := range buffer {
```

Comment:
Using variable name `segments` here looks confusing since we already have function parameter `segment`

Reply:
`segments` -> `bufferedSegments`
`segment` -> `newSegment`
`length` -> `numBuffered`

Reply:
Updated. 69d9ae1